File: broadcastrec/app/src/main/java/com/example/broadcastrec/MainActivity.java
package com.example.broadcastrec;


import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Toast;
import androidx.appcompat.app.AppCompatActivity;

/**
 * Activité principale pour démontrer l'utilisation d'un BroadcastReceiver.
 */
public class MainActivity extends AppCompatActivity {

    private BroadcastReceiver monReceiver;

    /**
     * Appelée lorsque l'activité est créée.
     */
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        // Définir un filtre pour le BroadcastReceiver
        IntentFilter filtre = new IntentFilter();
        filtre.addAction("com.data.CUSTOM_INTENT");

        // Initialisation du BroadcastReceiver
        monReceiver = new BroadcastReceiver() {
            @Override
            public void onReceive(Context context, Intent intent) {
                Toast.makeText(context, "Intent détecté", Toast.LENGTH_LONG).show();
                Log.d("tagReceived", "Intent reçu");
            }
        };

        // Enregistrer le BroadcastReceiver
        registerReceiver(monReceiver, filtre, Context.RECEIVER_NOT_EXPORTED);
    }

    /**
     * Méthode pour diffuser un intent particulier.
     */
    public void broadcastIntent(View view) {
        Intent intent = new Intent();
        intent.setAction("com.data.CUSTOM_INTENT");
        intent.setPackage(getPackageName()); // Cibler uniquement cette application
        sendBroadcast(intent);
        Log.d("tagBroadcast", "Broadcast envoyé");
    }

    /**
     * Nettoyer les ressources lorsque l'activité est détruite.
     */
    @Override
    public void onDestroy() {
        super.onDestroy();
        if (monReceiver != null) {
            unregisterReceiver(monReceiver);
            monReceiver = null;
        }
    }
}

File: MyApplication2/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.os.Bundle
import android.util.Log
import android.widget.Toast
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        Toast.makeText(this,"l'Activitee est cree",Toast.LENGTH_SHORT).show()
        Log.i("onCreate(): ","l'Activitee est cree")
    }

    override fun onStart(){
        super.onStart()
        Toast.makeText(this,"l'Activitee es demaree",Toast.LENGTH_SHORT).show()
        Log.i("onStart(): ","l'Activitee est demaree")
    }

    override fun onRestart() {
        super.onRestart()
        Toast.makeText(this,"l'Activitee es redemaree",Toast.LENGTH_SHORT).show()
        Log.i("onRestart(): ","l'Activitee est redemaree")
    }

    override fun onResume() {
        super.onResume()
        Toast.makeText(this,"l'Activitee est reprise",Toast.LENGTH_SHORT).show()
        Log.i("onResume(): ","l'Activitee est reprise")
    }

    override fun onPause() {
        super.onPause()
        Toast.makeText(this,"l'Activitee est en pause",Toast.LENGTH_SHORT).show()
        Log.i("onPause(): ","l'Activitee est en pause")
    }

    override fun onStop() {
        super.onStop()
        Toast.makeText(this,"l'Activitee est stopee",Toast.LENGTH_SHORT).show()
        Log.i("onStop(): ","l'Activitee est stopee")
    }

    override fun onDestroy() {
        super.onDestroy()
        Toast.makeText(this,"l'Activitee est detruite",Toast.LENGTH_SHORT).show()
        Log.i("onDestroy(): ","l'Activitee est detruite")
    }
}